packages/chat-component-bindings/src/hooks/useSelector.ts

```diff
@@ -2,30 +2,22 @@
 // Licensed under the MIT license.
 
 import { ChatClientState, StatefulChatClient } from 'chat-stateful-client';
-import { useChatClient } from '../providers/ChatClientProvider';
+import { ChatClientContext } from '../providers/ChatClientProvider';
 
-import { useState, useEffect, useRef, useMemo } from 'react';
-import { useChatThreadClient } from '../providers/ChatThreadClientProvider';
+import { useState, useEffect, useRef, useMemo, useContext } from 'react';
+import { ChatThreadClientContext } from '../providers/ChatThreadClientProvider';
 
 // This function highly depends on chatClient.onChange event
 // It will be moved into selector folder when the ChatClientProvide when refactor finished
-export const useSelector = <SelectorT extends (state: ChatClientState, props: any) => any>(
-  selector: SelectorT,
+export const useSelector = <
+  SelectorT extends (state: ChatClientState, props: any) => any,
+  ParamT extends SelectorT | undefined
+>(
+  selector: ParamT,
   selectorProps?: Parameters<SelectorT>[1]
-): ReturnType<SelectorT> => {
-  const chatClient: StatefulChatClient = useChatClient() as any;
-  const threadId = useChatThreadClient().threadId;
-
-  // Keeps track of whether the current component is mounted or not. If it has unmounted, make sure we do not modify the
-  // state or it will cause React warnings in the console. https://skype.visualstudio.com/SPOOL/_workitems/edit/2453212
-  const mounted = useRef(false);
```

Comment:
The `mounted` logic here shouldn't change at all?

Reply:
Interesting, looks like a problem of the merge, let me fix it